source/draw/area/area.py

```diff
@@ -56,25 +54,44 @@ def delete_entity_at_coordinates(self, entity, x, y) -> Entity:
             Entity: [description]
         """,
         try:
-            if (x,y) in self.entity_dict:
-                self.entity_dict[(x,y)].remove(entity)
+            if (entity.get_abs_x(), entity.get_abs_y()) in self.entity_dict:
+                self.entity_dict[(entity.get_abs_x(),entity.get_abs_y())].remove(entity)
                 entity.curr_area = None
-            return entity
+                return entity
         except ValueError:
             return None
     
-    #TODO: Add transfer and delete multi tile entity
-    def transfer_entity_between_coordinates(self, entity, x1, y1, x2, y2) -> None:
-        """Moves the entity at x1, y1 to x2, y2
+    def update_entity_at_coordinates(self, entity, old_x, old_y):
```

Comment:
Rename this to something like "move". Update sounds like we are changing the properties of the entity in-place, whereas the entity has actually been edited in the past. Also, I find it weird that we are passing the old coordinates on a changed entity instead of passing the new coordinates and having this function enact the change.

EDIT: Wait, right below this we have move_entity_to_coordinates which does exactly that (admittedly by calling this). Why are these two separate functions? What is the use case here? I feel like we should pick one pattern and standardize to it.

Reply:
Update is used because this function updates the position of the entity in the area dictionary to correspond to it's reported position.  In some sense, 'Update..." does not actually move the entity, just ensures the area dictionary knows where it is.

Update is called if an entity has moved in world space due to it's parent moving- for example, when the ship moves, it calls update on it's entity, which then tells the area dictionary what it's new coordinates  are. 

Move_... should be used if an entity needs to be repositioned independant of it's parent. There is a strong argument that this should be done in entity instead, but I think this does answer your immediate question.